### ISSUE TITLE
Question regarding block_in_if_condition_stmt

### ISSUE THREAD

**Issue by phimuemue** [rust-lang-nursery/rust-clippy#2784]
I regularly encounter a lint suggesting not to use blocks in conditions ([block_in_if_condition_stmt](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#block_in_if_condition_stmt)). While I realize that this lint may be useful in some situations, introducing a local variable seems inconvenient in many cases.

Could we make this lint configurable via `clippy.toml`, so that it can be disabled?

**Comment by oli-obk** [comment]
Do you have a few examples of where it's triggering for you?

**Comment by phimuemue** [comment]
As a (contrived) example, imagine the following code:
```
fn main() {
    let vec = vec![1,2,3,4];
    if vec.into_iter()
        .any(|i|
             2==i
        )
    {
        println!("Hello, world!");
    }
}
```

The above does not trigger this lint. But if I change the the lambda to a block (e.g. to add an `assert`), the lint triggers:

```
fn main() {
    let vec = vec![1,2,3,4];
    if vec.into_iter()
        .any(|i| {
             assert!(0<=i);
             2==i
        })
    {
        println!("Hello, world!");
    }
}
```

In my opinion, both alternatives are equally readable, and I would like to allow both of them in my code.

**Comment by oli-obk** [comment]
You can add `#![cfg_attr(cargo-clippy, allow(block_in_if_condition))]` to your main.rs in order to stop triggering the lint at all.

**Comment by phimuemue** [comment]
Thanks a lot! This helped.
